### PR TITLE
build: add dependabot.yml

### PR DESCRIPTION
**Pull request by lloydchang**
> [!IMPORTANT]
> Add `.github/dependabot.yml` to configure Dependabot with a weekly update schedule.
> 
>   - **Configuration**:
>     - Adds `.github/dependabot.yml` to configure Dependabot for version updates.
>     - Sets `version` to `2` and `schedule` to `weekly`.
>     - Leaves `package-ecosystem` unspecified, requiring future updates for specific ecosystems.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ErikBjare%2Fgptme&utm_source=github&utm_medium=referral)<sup> for 6f1387d476e1c7e13ba1bfde0651fed81d0528b2. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://docs.github.com/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file
+
+version: 2
+updates:
+  - package-ecosystem: "" # See documentation for possible values
```
[the remark this answers is not in the document]
Comment (lloydchang) [reply .github/dependabot.yml]: Suggested "npm" (for TypeScript, etc.) is inappropriate for this repository that uses Python.

To address the feedback, I added a new commit for "poetry":
https://github.com/ErikBjare/gptme/pull/274/commits/28ce28193edb23cae7d4b76629645337e173eac1

[the remark this answers is not in the document]
Comment (ErikBjare) [reply .github/dependabot.yml]: Seems the current value is `pip`: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://docs.github.com/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file
+
+version: 2
+updates:
+  - package-ecosystem: "poetry" # See documentation for possible values
```
Comment (ErikBjare) [inline .github/dependabot.yml]:
```suggestion
  - package-ecosystem: "pip"
```

**Comment by ErikBjare** [comment]
Thanks!

File: .github/dependabot.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (lloydchang) [reply .github/dependabot.yml]: Thanks!

**Pull request merged**